### PR TITLE
Revert reason scripts: remove duplicate code, improve error handling

### PR DESCRIPTION
**Pull request by RensR**
## Solution

### CODE REVIEW

**Review by PiotrTrzpil — APPROVED** [review]

File: core/scripts/ccip/ccip-revert-reason/main.go [hunk 1]
```diff
@@ -1,173 +1,63 @@
...
-		requester := "0xe88ff73814fb891bb0e149f5578796fa41f20242"
-
-		ec, ethErr := ethclient.Dial(ethUrl)
-		panicErr(ethErr)
-		errorString, _ := getErrorForTx(ec, txHash, requester)
-		// Some nodes prepend "Reverted " and we also remove the 0x
-		trimmed := strings.TrimPrefix(errorString, "Reverted ")[2:]
-
-		contractABIs := getAllABIs()
+	errorSting := ErrorCodeString
```
Comment (dimkouv) [inline core/scripts/ccip/ccip-revert-reason/main.go]: typo `errorString`

File: core/scripts/ccip/ccip-revert-reason/main.go [hunk 2]
```diff
@@ -1,173 +1,63 @@
...
-					errorBytes := v.([]interface{})[0].([]byte)
-					DecodeErrorStringFromABI(hex.EncodeToString(errorBytes), contractABIs)
-					return
-				}
-				fmt.Printf("Error is \"%v\" args %v\n", errorName, v)
-				return
-			}
+	if errorSting == "" {
+		// Try to load env vars from .env file
+		err := godotenv.Load(".env")
```
Comment (dimkouv) [inline core/scripts/ccip/ccip-revert-reason/main.go]: Maybe make `".env"` a const

File: core/scripts/ccip/ccip-revert-reason/main.go [hunk 3]
```diff
@@ -1,173 +1,63 @@
...
-func parseError(txError error) string {
-	b, err := json.Marshal(txError)
-	panicErr(err)
-	var callErr struct {
-		Code    int
-		Data    string `json:"data"`
-		Message string `json:"message"`
+	decodedError, err := handler.DecodeErrorStringFromABI(errorSting)
+	if err != nil {
+		panic(err)
```
Comment (dimkouv) [inline core/scripts/ccip/ccip-revert-reason/main.go]: I would suggest creating a separate function to run the `load error string` logic in `func main`

```
func main() {
   errorString, err := getErrorString()
   if err != nil -> panic
   
   decoded, err := decode(errorString)
   if err != nil -> panic
}

func getErrorString() {
   if err != nil -> return fmt.Errorf("get error for tx: %w", err)
}
```

File: core/scripts/ccip/secrets/secrets.go [hunk 4]
```diff
@@ -11,5 +11,5 @@ func GetRPC(chainID uint64) string {
 	if rpc != "" {
 		return rpc
 	}
-	panic("RPC not found. Please check secrets.go for chainID " + strconv.FormatUint(chainID, 10))
+	panic("RPC not found. Please set the environment variable for chain " + strconv.FormatUint(chainID, 10) + " e.g. RPC_420=https://rpc.420.com")
```
Comment (dimkouv) [inline core/scripts/ccip/secrets/secrets.go]: ```panic(fmt.Errorf("RPC not found. Please set the environment variable for chain %d e.g. RPC_420=https://rpc.420.com", chainID))```

**Review by dimkouv — APPROVED** [review]

**Pull request merged**